libs/vkd3d/memory.c

```diff
@@ -1280,7 +1305,7 @@ static HRESULT vkd3d_memory_allocation_init(struct vkd3d_memory_allocation *allo
     else
     {
         hr = vkd3d_allocate_device_memory(device, memory_requirements.size, type_flags,
-                type_mask, &flags_info, true, &allocation->device_allocation);
+                type_mask, &flags_info, info->heap_properties.Type == D3D12_HEAP_TYPE_UPLOAD, &allocation->device_allocation);
```

Comment:
I don't understand the rationable behind this line.

Reply:
Ignore the budget if we are allocating memory for GPU UPLOAD HEAP. In case the game only allocates GPU upload heap memory after allocating enough regular UPLOAD HEAP memory to use up all the budget.

Reply:
Right, this is the respect budget field, thought it was something about fallback ...

To make sure, CUSTOM heap with L0 heap will be normalized to UPLOAD here, right?

Reply:
I think we can assert here that heap property is no longer CUSTOM once we reach this code path.

Reply:
This ignores custom heap, so I changed it to `!is_host_visible_vram`.